src/serialization/IParseNode.cs

```diff
@@ -139,5 +139,10 @@ public interface IParseNode
         /// </summary>
         /// <returns>The byte array value of the node.</returns>
         byte[]? GetByteArrayValue();
+        /// <summary>
+        /// Gets the untyped value of the node
+        /// </summary>
+        /// <returns>The untyped value of the node.</returns>
+        UntypedNode? GetUntypedValue();
```

Comment:
the challenge with extending the interfaces is that it's a breaking change. We can leave this for now but we should think about a different design to avoid this breaking change, otherwise the changes will have to wait the next major version

Reply:
I'm able to serialize/deserialize untyped nodes without extending the interfaces.